search-service/src/main/kotlin/com/egm/stellio/search/web/BuildTemporalApiResponse.kt

```diff
@@ -0,0 +1,220 @@
+package com.egm.stellio.search.web
+
+import com.egm.stellio.search.model.TemporalEntitiesQuery
+import com.egm.stellio.search.model.TemporalQuery
+import com.egm.stellio.shared.model.CompactedAttribute
+import com.egm.stellio.shared.model.CompactedEntity
+import com.egm.stellio.shared.model.toFinalRepresentation
+import com.egm.stellio.shared.util.JsonUtils.serializeObject
+import com.egm.stellio.shared.util.buildQueryResponse
+import com.egm.stellio.shared.util.parseRepresentations
+import com.egm.stellio.shared.util.prepareGetSuccessResponseHeaders
+import com.egm.stellio.shared.util.toHttpHeaderFormat
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.util.MultiValueMap
+import java.time.ZonedDateTime
+
+typealias CompactedTemporalAttribute = List<Map<String, Any>>
+typealias TemporalValue = List<Any>
+const val TEMPORAL_VALUE_NAME = "values"
+
+typealias Range = Pair<ZonedDateTime, ZonedDateTime>
+
+object TemporalApiResponse {
+    @SuppressWarnings("LongParameterList")
+    fun buildListTemporalResponse(
+        entities: List<CompactedEntity>,
+        total: Int,
+        resourceUrl: String,
+        query: TemporalEntitiesQuery,
+        requestParams: MultiValueMap<String, String>,
+        mediaType: MediaType,
+        contexts: List<String>
+    ): ResponseEntity<String> {
+        val ngsiLdDataRepresentation = parseRepresentations(requestParams, mediaType)
+
+        val successResponse = buildQueryResponse(
+            entities.toFinalRepresentation(ngsiLdDataRepresentation),
+            total,
+            resourceUrl,
+            query.entitiesQuery.paginationQuery,
+            requestParams,
+            mediaType,
+            contexts
+        )
+
+        if (query.temporalQuery.asLastN) { // if lastN > limit it throw an error earlier
+            return successResponse
+        }
+
+        val attributesWhoReachedLimit = getAttributesWhoReachedLimit(entities, query)
```

Comment:
since there is now always a limit, how can an attribute go over it?

Reply:
they do not go over it but i need to know if some attribute actually reach the limit in order to trigger the pagination mechanism if one of them did.

Reply:
i get all of them since i use  them to calculate the range later.